### ISSUE TITLE
Master process crashed: Failed to run journal checkpoint thread, crashing.

### ISSUE THREAD

**Issue by witgo** [Alluxio/alluxio#8618]
**Alluxio Version:**
1.8.1

**Describe the bug**
```
java.lang.RuntimeException: alluxio.exception.FileDoesNotExistException: inodeId 10113188364287 does not exist; too many retries
	at alluxio.master.file.DefaultFileSystemMaster.completeFileFromEntry(DefaultFileSystemMaster.java:1294)
	at alluxio.master.file.DefaultFileSystemMaster.processJournalEntry(DefaultFileSystemMaster.java:478)
	at alluxio.master.journal.ufs.UfsJournalCheckpointThread.runInternal(UfsJournalCheckpointThread.java:146)
	at alluxio.master.journal.ufs.UfsJournalCheckpointThread.run(UfsJournalCheckpointThread.java:123)
Caused by: alluxio.exception.FileDoesNotExistException: inodeId 10113188364287 does not exist; too many retries
	at alluxio.master.file.meta.InodeTree.lockFullInodePath(InodeTree.java:393)
	at alluxio.master.file.DefaultFileSystemMaster.completeFileFromEntry(DefaultFileSystemMaster.java:1290)
	... 3 more

```
**To Reproduce**
Create a dir with the same name concurrently

**Additional context**
```

sequence_number: 5222362
inode_file {
  id: 10112383057919
  parent_id: 9
  name: "application_1537944329020_1020_1.inprogress"
  persistence_state: "PERSISTED"
  pinned: false
  creation_time_ms: 1553121254632
  last_modification_time_ms: 1553121254632
  block_size_bytes: 536870912
  length: 0
  completed: false
  cacheable: true
  ttl: -1
  owner: "xxx"
  group: "xxx"
  mode: 420
  ttlAction: DELETE
  ufs_fingerprint: ""
}
--------------------------------------------------------------------------------
sequence_number: 5222363
set_attribute {
  id: 10112383057919
  op_time_ms: 1553121255025
  ttl: -1
  permission: 420
  ttlAction: DELETE
}


sequence_number: 5222866
inode_file {
  id: 10113188364287
  parent_id: 9
  name: "application_1537944329020_1020_1.inprogress"
  persistence_state: "PERSISTED"
  pinned: false
  creation_time_ms: 1553121288532
  last_modification_time_ms: 1553121288532
  block_size_bytes: 33554432
  length: 0
  completed: false
  cacheable: true
  ttl: -1
  owner: "xxx"
  group: "xxx"
  mode: 438
  ttlAction: DELETE
  ufs_fingerprint: ""
}
--------------------------------------------------------------------------------
sequence_number: 5222867
complete_file {
  block_ids: 10113171587072
  id: 10113188364287
  length: 103889
  op_time_ms: 1553121283557
  ufs_fingerprint: "TYPE:FILE UFS:xxx OWNER:omm GROUP:omm MODE:438 CONTENT_HASH:(len_103889,_modtime_1553121283557) "
}



sequence_number: 5222904
rename {
  id: 10112383057919
  dst_path: "/spark/sparkJobHistory/application_1537944329020_1020_1"
  op_time_ms: 1553121288251
}


````

**Comment by apc999** [comment]
@witgo Thanks for reporting this issue. To repro this issue, do you mean "Create a dir with the same name concurrently" then restart Alluxio master? Because I saw the error message is about journal replaying? Also, do you see any exception when you "Create a dir with the same name concurrently"?

**Comment by witgo** [comment]
@apc999 The primary master node is not restarted and the log appears on secondary master .  no other exception information

**Comment by aaudiber** [comment]
@witgo could the first `"application_1537944329020_1020_1.inprogress"` (inode `10112383057919`) have been renamed before the second one (inode `10113188364287`) was created? There's a few hundred journal entries between the two inodes being created. Are there any other entries that reference `10112383057919`?

**Comment by aaudiber** [comment]
@witgo do you have a reliable way to reproduce the issue, or did this just happen one time?

**Comment by witgo** [comment]
@aaudiber   The rename event is the last one that happened. No other related events. This bug has appeared several times, but I don't know how to reproduce it.

**Comment by aaudiber** [comment]
@witgo We will aim to fix this in 1.8.2 once we determine the root cause. To help with debugging, can you share all entries in the range from sequence number `5222362` to sequence number `5222904`?

**Comment by witgo** [comment]
I'm sorry, I've erased it, but I don't think there's any other useful information in it.

**Comment by LuQQiu** [comment]
@witgo This journal related exceptions are thrown only when running Spark application or also happens when doing other alluxio integrations? If it is only thrown when running Spark applications, could you share us more information about this Spark application?

**Comment by LuQQiu** [comment]
@witgo Looks like the "application_1537944329020_1020_1.inprogress" is a log file when `spark.eventLog.enabled=true`. Did you enable the eventLog and set the `spark.eventLog.dir` to alluxio address?

**Comment by witgo** [comment]
@LuQQiu   We only use alluxio in spark scenarios.   yes,enable the eventLog and set the spark.eventLog.dir to alluxio address.  There are no special places other than frequent start-stop spark apps on yarn.

**Comment by LuQQiu** [comment]
This issue is able to reproduce by 50 threads continually issue one of the four commands: 
`createFile(src)`, `completeFile(src)`, `rename(src, dst)`, `delete(dst)`.

The journal sequence may be different from the actual execution sequence.

The `FileDoesNotExistException: inodeId 10113188364287 does not exist` issue happens when we have the following journal sequence 
```
id1:create(a) -> id1:complete (a) -> id2:create(a) -> id2:complete(a) -> id1: rename(a -> b)
```

The `FileDoesNotExistException: Path "a" does not exist` issue happens when we have the following journal sequence
```
id1:create(a) -> id1:complete (a) -> id2:create(a) -> id1: rename(a -> b) ->  id2:complete(a) 
```
There may be some locking issues in our rename process and we are working to solve the issue.

**Comment by LuQQiu** [comment]
@calvinjia This issue is fixed in #8768, Please help close this issue
